social_protection/services.py

```diff
@@ -0,0 +1,56 @@
+import logging
+
+from django.core.exceptions import ValidationError
+
+from core.services import BaseService
+from core.signals import register_service_signal
+from social_protection.models import BenefitPlan
+from social_protection.validation import BenefitPlanValidation
+
+logger = logging.getLogger(__name__)
+
+
+class BenefitPlanService(BaseService):
+    @register_service_signal('benefit_plan_service.create')
+    def create(self, obj_data):
+        incoming_code = obj_data.get('code')
+        if check_unique_code(incoming_code):
+            raise ValidationError(("Benefit code %s already exists" % incoming_code))
+        incoming_name = obj_data.get('name')
+        if check_unique_name(incoming_name):
+            raise ValidationError(("Benefit name %s already exists" % incoming_name))
```

Comment:
Use validation class to perform this checks (for both create and update).

Reply:
done